src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1000,6 +1006,36 @@ class DocBaseViewer extends BaseViewer {
             // Add page IDs to each page after page structure is available
             this.setupPageIds();
         }
+
+        if (this.options.enableThumbnailsSidebar) {
+            this.initThumbnails();
+        }
+    }
+
+    initThumbnails() {
+        this.thumbnailsSidebar = new VirtualScroller(this.thumbnailsSidebarEl);
+
+        // Get the first page of the document, and use its dimensions
+        // to set the thumbnails size of the thumbnails sidebar
+        this.pdfViewer.pdfDocument.getPage(1).then((page) => {
+            const desiredWidth = DEFAULT_THUMBNAILS_SIDEBAR_WIDTH;
+            const viewport = page.getViewport(1);
+            const scale = desiredWidth / viewport.width;
```

Comment:
Is it possible for this to return undefined or zero? Can `viewport` be undefined?